src/world/footpath.c

```diff
@@ -473,20 +474,197 @@ void footpath_provisional_update()
 	footpath_provisional_remove();
 }
 
+void sub_689726_helper(uint16 *ax, uint16 *bx, uint16 *cx, uint16 *dx, uint32 rotation)
```

Comment:
My thought was that it was easier to first free ourselves from the rct2 black box and then refactor code in something that makes more sense. However, I shall go over it tonight to improve it a bit.

And thank you for that function. (Although I don't get what you mean about the rotation variable.) I think it was a macro in the original assembly, because I derived `sub_689726_helper` from 0x689726 itself.